File: psana_examples/tags/V00-04-11/src/dump_partition.py
#--------------------------------------------------------------------------
# File and Version Information:
# 	$Id:
#
# Description:
#	Class DumpPartition...
#
# Author List:
#
#------------------------------------------------------------------------

#--------------------------------
#  Imports of standard modules --
#--------------------------------
import sys
import logging

#-----------------------------
# Imports for other modules --
#-----------------------------
from psana import *

#----------------------------------
# Local non-exported definitions --
#----------------------------------

def _printPartition(config):
    print "Partition::ConfigV1:"
    print "  bldMask: 0x%16.16x" % config.bldMask()
    print "  numSources: %d" % config.numSources()
    sources = config.sources()
    for source in sources:
        src = source.src()
        group = source.group()
        print "    src= %s group= %s" % (src,group)
    
#---------------------
#  Class definition --
#---------------------
class dump_partition (object) :
    '''Class whose instance will be used as a user analysis module.'''

    #----------------
    #  Constructor --
    #----------------
    def __init__(self):
        '''Class constructor. Does not need any parameters'''
        pass

    #-------------------
    #  Public methods --
    #-------------------

    def beginJob( self, evt, env ) :
        """This optional method is called if present at the beginning 
        of the new run.

        @param evt    event data object
        @param env    environment object
        """

        config = env.configStore().get(Partition.ConfigV1, Source("ProcInfo()"))
        if config:
            print "dump_partition in beginJob()"
            _printPartition(config)


    def event( self, evt, env ) :
        """This method is called for every L1Accept transition.

        @param evt    event data object
        @param env    environment object
        """
        pass

File: psana_examples/tags/V00-05-01/src/DumpPartition.cpp
//--------------------------------------------------------------------------
// File and Version Information:
// 	$Id:
//
// Description:
//	Class DumpPartition...
//
// Author List:
//
//------------------------------------------------------------------------

//-----------------------
// This Class's Header --
//-----------------------
#include "psana_examples/DumpPartition.h"

//-----------------
// C/C++ Headers --
//-----------------
#include <iomanip>
#include <iostream>

//-------------------------------
// Collaborating Class Headers --
//-------------------------------
#include "MsgLogger/MsgLogger.h"
#include "psddl_psana/partition.ddl.h"

//-----------------------------------------------------------------------
// Local Macros, Typedefs, Structures, Unions and Forward Declarations --
//-----------------------------------------------------------------------

using namespace psana_examples;
PSANA_MODULE_FACTORY(DumpPartition)

namespace {
  void printPartition(std::ostream &out, boost::shared_ptr<Psana::Partition::ConfigV1> config) {
    out << "Partition::ConfigV1:\n";
    out << "  bldMask: " <<  "0x" << std::setw(16) << std::setfill('0') << std::hex << config->bldMask() << '\n';
    out << "  numSources: " << config->numSources() << '\n';
    ndarray<const Psana::Partition::Source, 1> sources = config->sources();
    for (unsigned idx = 0; idx < config->numSources(); ++ idx) {
      const Psana::Partition::Source & source = sources[idx];
      const Pds::Src& src = source.src();
      uint32_t group = source.group();
      out << "    src= " << src << "  group= " << group << '\n';
    }
  }
}

//		----------------------------------------
// 		-- Public Function Member Definitions --
//		----------------------------------------

namespace psana_examples {

//----------------
// Constructors --
//----------------
DumpPartition::DumpPartition (const std::string& name)
  : Module(name)
{
}

//--------------
// Destructor --
//--------------
DumpPartition::~DumpPartition ()
{
}

void 
DumpPartition::beginJob(Event& evt, Env& env)
{
  PSEvt::Source src("ProcInfo()");
  boost::shared_ptr<Psana::Partition::ConfigV1> config = env.configStore().get(src);
  if (config) {
    MsgLog(name(),info,"DumpPartition In beginJob()");
    printPartition(std::cout, config);
  }
}

// Method which is called with event data
void 
DumpPartition::event(Event& evt, Env& env)
{
}
  
} // namespace psana_examples
